docs/csharp/tutorials/console-webapiclient.md

```diff
@@ -1,388 +1,354 @@
 
-Before you add more features, let's address the `name` property by using the `[JsonPropertyName]` attribute. Make
-the following changes to the declaration of the `name` field in repo.cs:
+   The JSON for a repository object contains dozens of properties,but only the `name` property will be deserialized. The serializer automatically ignores JSON properties for which there is no match in the target class. This feature makes it easier to create types that work with only a subset of fields in a large JSON packet.
 
-```csharp
-[JsonPropertyName("name")]
-public string Name { get; set; }
-```
+   The C# convention is to capitalize the first letter of property names, but the `name` property here starts with a lowercase letter because that matches exactly what's in the JSON. Later you'll see how to use C# property names that don't match the JSON property names.
 
-To use `[JsonPropertyName]` attribute, you will need to add the <xref:System.Text.Json.Serialization> namespace to the `using` directives:
+1. Use the serializer to convert JSON into C# objects. Replace the call to
+<xref:System.Net.Http.HttpClient.GetStringAsync(System.String)> in the `ProcessRepositories` method with the following lines:
 
-```csharp
-using System.Text.Json.Serialization;
-```
+   ```csharp
+   var streamTask = client.GetStreamAsync("https://api.github.com/orgs/dotnet/repos");
+   var repositories = await JsonSerializer.DeserializeAsync<List<Repository>>(await streamTask);
+   ```
 
-This change means you need to change the code that writes the name of each repository in program.cs:
+   The new code calls <xref:System.Net.Http.HttpClient.GetStreamAsync(System.String)> instead of <xref:System.Net.Http.HttpClient.GetStringAsync(System.String)>. This serializer method uses a stream instead of a string as its source.
 
-```csharp
-Console.WriteLine(repo.Name);
-```
+   The first argument to <xref:System.Text.Json.JsonSerializer.DeserializeAsync%60%601(System.IO.Stream,System.Text.Json.JsonSerializerOptions,System.Threading.CancellationToken)?displayProperty=nameWithType> is an `await` expression. (The other two parameters are optional and are omitted in the code snippet.) `await` expressions can appear almost anywhere in your code, even though up to now, you've only seen them as part of an assignment statement.
 
-Execute `dotnet run` to make sure you've got the mappings correct. You should
-see the same output as before.
+   The `DeserializeAsync` method is *generic*, which means you supply type arguments for what kind of objects should be created from the JSON text. In this example, you're deserializing to a `List<Repository>`, which is another generic object, a <xref:System.Collections.Generic.List%601?displayProperty=nameWithType>. The `List<T>` class stores a collection of objects. The type argument declares the type of objects stored in the `List<T>`. The type argument is your `Repository` class, because the JSON text represents a collection of repository objects.
```

Comment:
link to https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/generics/generic-methods or better https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/generics/

That's not what generic means, but generic allows - Right?

Reply:
>That's not what generic means, but generic allows - Right?

In this case it's more than just allows, as the type parameter is required.